viite-backend/viite-main/src/main/scala/fi/vaylavirasto/viite/dynamicnetwork/DynamicRoadNetworkService.scala

```diff
@@ -373,71 +373,96 @@ class DynamicRoadNetworkService(linearLocationDAO: LinearLocationDAO, roadwayDAO
-      val roadway = roadwayDAO.fetchByRoadwayNumber(roadwayNumber).getOrElse(throw ViiteException(s"No Roadway found for linear location id: ${change.oldLinkId} startM: ${change.oldStartM} endM: ${change.oldEndM}"))
-      val otherRoadwayNumbers = linLocsGroupedByLinkId.filterNot(grp => grp._1 == change.oldLinkId).map(_._2.map(_.roadwayNumber)).toSeq.flatten.toSet
-      val otherRoadways = roadwayDAO.fetchAllByRoadwayNumbers(otherRoadwayNumbers)
+      val roadwayNumber = filterByLinkIdAndMValueRange(linearLocations, change.oldLinkId, change.oldStartM, change.oldEndM).head.roadwayNumber
+      val roadway = roadwaysForLinearLocations.find(_.roadwayNumber == roadwayNumber).getOrElse(throw ViiteException(s"No Roadway found for linear location id: ${change.oldLinkId} startM: ${change.oldStartM} endM: ${change.oldEndM}"))
+      val otherRoadwayNumbers = linLocsGroupedByLinkId.filterNot(grp => grp._1 == change.oldLinkId).map(_._2.map(_.roadwayNumber)).toSet.flatten
+      val otherRoadways = roadwaysForLinearLocations.filter(rw => otherRoadwayNumbers.contains(rw.roadwayNumber)).toSet
       existsConnectingRoadway(roadway, otherRoadways)
     }
 
-    def getMetaData(change: TiekamuRoadLinkChange, activeLinearLocations: Seq[LinearLocation]): TiekamuRoadLinkErrorMetaData = {
+    def getMetaData(change: TiekamuRoadLinkChange, activeLinearLocations: Seq[LinearLocation], roadwaysForLinearLocations: Set[Roadway]): TiekamuRoadLinkErrorMetaData = {
       val errorLink = change.oldLinkId
       val errorLinearLocations = activeLinearLocations.filter(ll => ll.linkId == errorLink)
-      val errorRoadwayNumbers = errorLinearLocations.map(_.roadwayNumber)
-      val errorRoadways = roadwayDAO.fetchAllByRoadwayNumbers(errorRoadwayNumbers.toSet)
+      val errorRoadwayNumbers = errorLinearLocations.map(_.roadwayNumber).toSet
+      val errorRoadways = roadwaysForLinearLocations.filter(rw => errorRoadwayNumbers.contains(rw.roadwayNumber)).toSet
       val errorRoadsParts = errorRoadways.map(r => {
         (r.roadPart)
       })
       TiekamuRoadLinkErrorMetaData(errorRoadsParts.head, errorRoadways.head.roadwayNumber, errorLinearLocations.map(_.id), errorLink)
     }
 
-    def validateCombinationCases(otherChangesWithSameNewLinkId: Seq[TiekamuRoadLinkChange], change: TiekamuRoadLinkChange, linearLocations: Seq[LinearLocation]): Seq[TiekamuRoadLinkChangeError] = {
-      def areHomogeneous(linearLocations: Seq[LinearLocation]): Boolean = {
-        val roadwayNumbers = linearLocations.map(_.roadwayNumber).toSet
-        val roadways = roadwayDAO.fetchAllByRoadwayNumbers(roadwayNumbers)
+    def validateCombinationCases(otherChangesWithSameNewLinkId: Seq[TiekamuRoadLinkChange], change: TiekamuRoadLinkChange, linearLocations: Seq[LinearLocation], roadwaysForLinearLocations: Set[Roadway]): Seq[TiekamuRoadLinkChangeError] = {
+      def areHomogeneous(linearLocations: Set[LinearLocation]): Boolean = {
```

Comment:
(Vähän out-of scope itse puollarista, mut tähän kaipaisin selityksen tolle homogeenisyydelle - tai enemmänkin sille, miksei täs määritelmään kuulu elyä tai hallinnollista luokkaa)

Reply:
Tai jos ei ole varmaa tietoa, miksi tässä juuri tällainen homogeenisyyden määritelmä tässä, niin kommentti, jossa mainitaan esim., että hallinnollisen luokan lisäämiselle homogeenisyyskriteereihin saattaa olla tarpeen.